OAuth2 flow fails against docker trusted registry (missing client_id)

**Description of the issue**:

When pulling/pushing to a docker trusted registry, authentication fails with error: 
400 Bad Request {"details":"invalid client ID"}

Docker registry (DTR) requires a client_id parameter to be passed with the refresh_token when making a login request.

From
https://docs.docker.com/registry/spec/auth/oauth/

> client_id
>(REQUIRED) String identifying the client. This client_id does not need to be registered with the authorization server but should be set to a meaningful value in order to allow auditing keys created by unregistered clients. Accepted syntax is defined in [RFC6749 Appendix A.1](https://tools.ietf.org/html/rfc6749#appendix-A.1)

**Expected behavior**:
It should be possible to pull a base image from an internal docker registry (DTR) using OAuth2 authentication flow.

**Steps to reproduce**: 
* Modify base image to pull from a docker trusted registry (https://docs.docker.com/datacenter/dtr/2.1/guides/)
* Login to registry
Observe that the login is stored as an OAuth token
```
docker-credential-wincred list
{"internalregistry":"\u003ctoken\u003e"}
```

* Run ```mvn jib:dockerBuild```
Build fails with error:
Unauthorized for internalregistry/base/custom-base-image: 400 Bad Request
[ERROR] {"details":"invalid client ID"}


**Environment**:
jib-maven-plugin 1.0.2
maven 3.5.0
jdk 1.8.0_131

**`jib-maven-plugin` Configuration**: 
```xml
<from>
   <image>internalregistry/base/custom-base-image</image>
</from>
```

**Log output**: 
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.0.2:dockerBuild (default-cli) on project my-project Build to Docker daemon failed, perhaps you should set a credential helper name with the configuration '<from><credHelper>' or set credentials for 'internalregistry' in your Maven settings: Unauthorized for internalregistry/base/custom-base-image: 400 Bad Request
[ERROR] {"details":"invalid client ID"}

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
@@ -269,7 +269,7 @@ public class RegistryAuthenticator {
     String serviceScope = getServiceScopeRequestParameters(scope);
     return isOAuth2Auth()
         ? serviceScope
-            + "&grant_type=refresh_token&refresh_token="
+            + "&client_id=jib&grant_type=refresh_token&refresh_token="
             // If OAuth2, credential.getPassword() is a refresh token.
             + Verify.verifyNotNull(credential).getPassword()
         : serviceScope;

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
@@ -270,7 +270,7 @@ public class RegistryAuthenticator {
     return isOAuth2Auth()
         ? serviceScope
             // https://github.com/GoogleContainerTools/jib/pull/1545
-            + "&client_id=da031fe481a93ac107a95a96462358f9"
+            + "&client_id=jib.da031fe481a93ac107a95a96462358f9"
             + "&grant_type=refresh_token&refresh_token="
             // If OAuth2, credential.getPassword() is a refresh token.
             + Verify.verifyNotNull(credential).getPassword()

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java
@@ -269,7 +269,9 @@ public class RegistryAuthenticator {
     String serviceScope = getServiceScopeRequestParameters(scope);
     return isOAuth2Auth()
         ? serviceScope
-            + "&client_id=jib&grant_type=refresh_token&refresh_token="
+            // https://github.com/GoogleContainerTools/jib/pull/1545
+            + "&client_id=da031fe481a93ac107a95a96462358f9"
+            + "&grant_type=refresh_token&refresh_token="
             // If OAuth2, credential.getPassword() is a refresh token.
             + Verify.verifyNotNull(credential).getPassword()
         : serviceScope;

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
@@ -70,7 +70,7 @@ public class RegistryAuthenticatorTest {
     registryAuthenticator.setCredential(Credential.basic("<token>", "oauth2_access_token"));
     Assert.assertEquals(
         "service=someservice&scope=repository:someimage:scope"
-            + "&grant_type=refresh_token&refresh_token=oauth2_access_token",
+            + "&client_id=jib&grant_type=refresh_token&refresh_token=oauth2_access_token",
         registryAuthenticator.getAuthRequestParameters("scope"));
   }
 

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
@@ -70,7 +70,7 @@ public class RegistryAuthenticatorTest {
     registryAuthenticator.setCredential(Credential.basic("<token>", "oauth2_access_token"));
     Assert.assertEquals(
         "service=someservice&scope=repository:someimage:scope"
-            + "&client_id=da031fe481a93ac107a95a96462358f9"
+            + "&client_id=jib.da031fe481a93ac107a95a96462358f9"
             + "&grant_type=refresh_token&refresh_token=oauth2_access_token",
         registryAuthenticator.getAuthRequestParameters("scope"));
   }

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryAuthenticatorTest.java
@@ -70,7 +70,8 @@ public class RegistryAuthenticatorTest {
     registryAuthenticator.setCredential(Credential.basic("<token>", "oauth2_access_token"));
     Assert.assertEquals(
         "service=someservice&scope=repository:someimage:scope"
-            + "&client_id=jib&grant_type=refresh_token&refresh_token=oauth2_access_token",
+            + "&client_id=da031fe481a93ac107a95a96462358f9"
+            + "&grant_type=refresh_token&refresh_token=oauth2_access_token",
         registryAuthenticator.getAuthRequestParameters("scope"));
   }
 

